crates/bindings-csharp/Codegen/Module.cs

```diff
@@ -129,17 +160,31 @@ public TableDeclaration(GeneratorAttributeSyntaxContext context)
             throw new InvalidOperationException("Tagged enums cannot be tables.");
         }
 
-        var attrArgs = context.Attributes.Single().NamedArguments;
+        Visibility = context.TargetSymbol.DeclaredAccessibility switch {
+            Accessibility.ProtectedAndInternal
+            or Accessibility.NotApplicable
+            or Accessibility.Internal => "internal",
+            Accessibility.Public => "public",
+            _ => throw new Exception(
+                "Table row type visibility must be public or internal."
+            ),
+        };
 
-        IsPublic = attrArgs.Any(pair => pair is { Key: "Public", Value.Value: true });
+        Views = new(context.Attributes
+            .Where(a => a.AttributeClass?.ToDisplayString() == "SpacetimeDB.TableAttribute")
+            .Select(a => new TableView(this, a))
+            .ToImmutableArray());
+
+        var schedules = Views.Where(t => t.Scheduled != null).Select(t => t.Scheduled);
+        var numSchedules = schedules.Count();
```

Comment:
This will cause the iterable to be fully walked over twice.

Looks like the logic could be simplified if you just kept the nullable `t.Scheduled` before doing `.Distinct()` - it would catch both `null` as well as some non-null values as individual items, so you'd only need to do one check instead of several.

Reply:
This code still looks suspicious (maybe more so than last time?).

Now you have `numSchedules = schedules.Count();`, but because `schedules` is just a `Views.Select(...)`, you're essentially doing `numSchedules = Views.Length`.

And then inside the if condition you're checking `numSchedules != Views.Length`, which can never be true...